components/raftstore/src/store/util.rs

```diff
@@ -836,6 +837,153 @@ impl Display for MsgType<'_> {
     }
 }
 
+/// `RegionReadProgress` is used to keep track of the replica's `safe_ts`, the replica can handle a read
+/// request directly without requiring leader lease or read index iff `safe_ts` >= `read_ts` (the `read_ts`
+/// is usually stale i.e seconds ago).
+///
+/// `safe_ts` is forward by the `(apply index, safe ts)` item:
```

Comment:
This doesn't seem like a common usage. Do you have any recommendations? /cc @WT-Liu

Reply:
In this context, "update" might be more appropriate.